### DISCUSSION TITLE
strptime support for non zero-padded values?

### DISCUSSION THREAD

**Discussion by archetyped** [johnkerl/miller#1197] [Q&A] [answered]
Hi, I'm running into an issue when attempting to parse non zero-padded values with `strptime`:

```bash
echo d='1/6/2022' | mlr --opprint put '$ts = strptime($d,"%m/%d/%Y")'
```
Output
```bash
d        ts
1/6/2022 (error)
```

Support for non zero-padded values with the standard `strptime` format strings appears to be platform-specific, but I cannot find any details on whether they are supported in Go or Miller.

If non zero-padded values are not supported in Miller's `strptime` implementation, is there a different approach for handling such date formats?

Thanks.

**Comment by aborruso** [comment, marked as answer]
Hi @archetyped I have no satisfactory answer for you. But I add a note for @johnkerl : using Miller 5, it works.

**Comment by johnkerl** [comment]
This is a bug: #1199

**Comment by archetyped** [comment]
Thanks @aborruso and @johnkerl for looking into this.

In the meantime, I'm currently dealing with non zero-padded date values with a simple regex replacement:

```bash
gsub($d, "\b(\d)\b", "0\1")
```
So the steps for parsing the date looks like:

```bash
echo d='1/6/2022' | \
mlr --opprint --barred put \
'$d_fix = gsub($d, "\b(\d)\b", "0\1");
$ts = strptime($d_fix,"%m/%d/%Y");
$d_fmt = strftime($ts, "%F")'
```
Output
```
+----------+------------+------------+------------+
| d        | d_fix      | ts         | d_fmt      |
+----------+------------+------------+------------+
| 1/6/2022 | 01/06/2022 | 1641427200 | 2022-01-06 |
+----------+------------+------------+------------+
```

More fragile than actual parsing, but it works in the interim.  I'd love to hear of any cleaner workarounds that others can think of.
